// Copyright 2024 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package pathology.orchestrator.spanner;

option java_package = "com.pathology.orchestrator.spanner";
option java_outer_classname = "SpannerResourcesProto";
option java_multiple_files = true;

// StoredPathologyOperationMetadata provides details of the Operation
// that is stored in Spanner and maps to PathologyOperationMetadata.
message StoredPathologyOperationMetadata {
  oneof metadata {
    // TransferDeid operation metadata.
    TransferDeidOperationMetadata transfer_deid_metadata = 1;

    // Export operation metadata.
    ExportOperationMetadata export_metadata = 2;
  }

  // Path in Cloud Storage where filter file is stored.
  string filter_file_path = 3;
}

// TransferDeidOperationMetadata provides details of a running
// TransferDeidPathologyCohort operation.
message TransferDeidOperationMetadata {
  // Resource name of new cohort generated.
  string resource_name_transferred = 1;
}

// ExportOperationMetadata provides details of a running
// ExportPathologyCohort operation.
message ExportOperationMetadata {
  // Cloud Storage bucket that data was exported to.
  string gcs_dest_path = 2;
}
